06 Faction ID Patch Fixes/gamedata/scripts/factionID_hud_mcm.script

```diff
@@ -0,0 +1,567 @@
+				{type = "title", text = ""},
+				{id = "xp"	, type = "track", val = 2, def = 0.0, 	min = -300.0, max = 300.0, 	step = 0.5},
+				{id = "yp"	, type = "track", val = 2, def = 0.0, 	min = -300.0, max = 300.0, 	step = 0.5},
+				{type = "title", text = ""},
+				{id = "xn"	, type = "track", val = 2, def = 16.0, 	min = -300.0, max = 300.0, 	step = 0.5},
+				{id = "yn"	, type = "track", val = 2, def = -14.0, min = -300.0, max = 300.0, 	step = 0.5},
+				{type = "title", text = ""},
+				{id = "xr"	, type = "track", val = 2, def = 16.0, 	min = -300.0, max = 300.0, 	step = 0.5},
+				{id = "yr"	, type = "track", val = 2, def = 3.0, 	min = -300.0, max = 300.0, 	step = 0.5},
+				{type = "line"},
+				
+				{type = "title", text = "Text"},
+				{id = "flipAlign" , type = "check", val = 1, def = false},
+				{id = "colorR"	, type = "track", val = 2, def = 200.0, 	min = 0.0, max = 255.0, step = 0.5},
+				{id = "colorG"	, type = "track", val = 2, def = 200.0, 	min = 0.0, max = 255.0, step = 0.5},
+				{id = "colorB"	, type = "track", val = 2, def = 200.0, 	min = 0.0, max = 255.0, step = 0.5},
+				{type = "title", text = ""},
+				{id = "RcolorR"	, type = "track", val = 2, def = 150.0, 	min = 0.0, max = 255.0, step = 0.5},
+				{id = "RcolorG"	, type = "track", val = 2, def = 150.0, 	min = 0.0, max = 255.0, step = 0.5},
+				{id = "RcolorB"	, type = "track", val = 2, def = 150.0, 	min = 0.0, max = 255.0, step = 0.5},
+				
+			}
+		}
+		
+	return op
+end
+
+local idtable = {}
+local nametable = {}
+local name_unknown = game.translate_string("ui_mcm_targetID_unknown")
+
+function identify(obj)
+	if not timeID then 
+		idtable[obj:character_name()] = 1
+		return true 
+	end
+	
+	if not idtable[obj:character_name()] then idtable[obj:character_name()] = 0 end
+	
+	if idtable[obj:character_name()] >= 1 then
+		return true
+	end
+
+	local distance = db.actor:position():distance_to_sqr(obj:position())
+	
+	if distance < 2500 then
+		idtable[obj:character_name()] = 1
+		return true
+	end
+	
+	local power = (100 / ((distance - 2500) * 0.1) / device().fov) * (tickSpeed * 0.1)
+	
+	--printf(distance .. " " .. power)
+	
+	idtable[obj:character_name()] = idtable[obj:character_name()] + (power * speedID)
+	if idtable[obj:character_name()] >= 1 then
+		return true
+	end
+	
+	return false
+end
+
+
+function displayFaction(obj)
+	if not obj then return end
+	
+	local fact = obj:character_community()
+	local name = obj:character_name()
+--	local rank = ranks.get_obj_rank_name(obj)
+--	rank = rank:sub(1,1):upper()..rank:sub(2)
+	rank = game.translate_string("st_rank_" .. ranks.get_obj_rank_name(obj)) or ""
+
+	if not rankID then rank = nil end
+	if not nameID then name = nil end
+	
+	if nameID and talkName and not nametable[name] then 
+		name = name_unknown
+		rank = nil
+	end
+
+	if deadID and obj.health <= 0 then
+		fact = fact .. "_dead"
+	end
+
+	display(fact, name, rank)
+end
+
+function display(faction, name, rank)
```

Comment:
This is the only function I modify for DART. I tried the monkey patch method of `function factionID_hud_mcm.display` to monkey patch this function but it did not take. I think it might be because of its dependency on variables